extensions/resource-deployment/src/ui/wizardBase.ts

```diff
@@ -26,8 +30,25 @@ export abstract class WizardBase<T, P extends WizardPageBase<T>, M extends Model
 		return this._useGenerateScriptButton;
 	}
 
-	constructor(private title: string, name: string, private _model: M, public toolsService: IToolsService, private _useGenerateScriptButton: boolean = false) {
-		this.wizardObject = azdata.window.createWizard(title, name);
+	public get resourceType(): ResourceType {
```

Comment:
As discussed offline, the wizard should just be passing these to the toolsAndEula page no other page/wizard should have to deal with these, so do no expose these getter/setters.